scenes/SceneEditAccountV2.js

```diff
@@ -0,0 +1,411 @@
+          alert: {
+            message: "Something went wrong with the upload. Please try again",
+          },
+        },
+      });
+      return;
+    }
+
+    // NOTE(jim): Only allow images for account avatar.
+    if (!file.type.startsWith("image/")) {
+      dispatchCustomEvent({
+        name: "create-alert",
+        detail: {
+          alert: { message: "Upload failed. Only images and gifs are allowed" },
+        },
+      });
+      return;
+    }
+
+    const response = await FileUtilities.upload({ file });
+
+    if (!response) {
+      dispatchCustomEvent({
+        name: "create-alert",
+        detail: {
+          alert: { message: "We're having trouble connecting right now" },
+        },
+      });
+      this.setState({ changingAvatar: false });
+      return;
+    }
+
+    if (response.error) {
+      dispatchCustomEvent({
+        name: "create-alert",
+        detail: { alert: { decorator: json.decorator } },
+      });
+      this.setState({ changingAvatar: false });
+      return;
+    }
+
+    const { json } = response;
+    const cid = json.data.ipfs.replace("/ipfs/", "");
+    const url = Strings.getCIDGatewayURL(cid);
+    await Actions.updateViewer({
+      data: {
+        photo: Strings.getCIDGatewayURL(cid),
+        body: this.state.body,
+        name: this.state.name,
+      },
+    });
+
+    await this.props.onRehydrate();
+    this.setState({ changingAvatar: false, photo: url });
+  };
+
+  _handleSaveAll = async (e) => {
+    this.setState({ changingUsername: true });
+    this.setState({ changingFilecoin: true });
+    this.setState({ changingBio: true });
+    this.setState({ changingUsername: true });
+    
+    if(this.state.password !== ""){
+      this.setState({ changingPassword: true });
+      
+     if (this.state.password !== this.state.confirm) {
+       dispatchCustomEvent({
+         name: "create-alert",
+         detail: { alert: { message: "Passwords did not match" } },
+       });
+       this.setState({ changingPassword: false });
+       return;
+     }
+  
+     if (!Validations.password(this.state.password)) {
+       dispatchCustomEvent({
+         name: "create-alert",
+         detail: {
+           alert: { message: "Password length must be more than 8 characters" },
+         },
+       });
+       this.setState({ changingPassword: false });
+       return;
+     }
+   }
+    await Actions.updateViewer({
+      username: this.state.username,
+      type: "CHANGE_PASSWORD",
+            password: this.state.password,
+      data: {
+        photo: this.state.photo,
+        body: this.state.body,
+        name: this.state.name,
+        allow_filecoin_directory_listing: this.state.allow_filecoin_directory_listing,
+        allow_automatic_data_storage: this.state.allow_automatic_data_storage,
+        allow_encrypted_data_storage: this.state.allow_encrypted_data_storage,
+      },
+    });
+
+    await this.props.onRehydrate();
+    this.setState({ changingFilecoin: false });
```

Comment:
This can all be in a single set start. They don't have to be in separate set states.